src/utils.py

```diff
@@ -13,31 +17,54 @@
     },
 }
 
-def print_options_at_distance(d, words, width, max_length, dmv, use_emoji):
+@dataclass
+class Transform:
+    cost: int
+
+@dataclass(order=True, frozen=True)
+class PrioritizedItem:
+    priority: int
+    item: Any=field(compare=False)
+
+@dataclass(order=True, frozen=True)
+class Option:
```

Comment:
nit - Really don't like Option as the name of this class - feels too close to typing.Optional particularly when it's in type annotations. CandidateWord?